src/libs/actions/Report.js

```diff
@@ -450,13 +451,21 @@ function openReport(reportID, participantLoginList = [], newReportObject = {}, p
                 avatar: UserUtils.getDefaultAvatarURL(accountID),
                 displayName: login,
             };
+
+            failurePersonalDetails[accountID] = allPersonalDetails[accountID] || null;
         });
         onyxData.optimisticData.push({
             onyxMethod: Onyx.METHOD.MERGE,
             key: ONYXKEYS.PERSONAL_DETAILS_LIST,
             value: optimisticPersonalDetails,
         });
 
+        onyxData.failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            value: failurePersonalDetails,
+        });
```

Comment:
Instead of setting the empty data in failureData for reportID, we can Clear the optimistic personal detail in deleteReport; clearing the report data early will cause the header to loading continuously and cause this [issue](https://github.com/Expensify/App/issues/38879)